lib/document_table.js

```diff
@@ -7,27 +7,31 @@ var ArgTypes = require("./arg_types");
 var DA = require("deasync");
 
 //Searching query for jsonb docs
-exports.searchDoc = function(args, next){
-  assert(args.keys && args.term, "Need the keys to use and the term string");
-
+exports.searchDoc = function(){
+  var args = ArgTypes.searchArgs(arguments, this);
+  assert(args.fields.keys && args.fields.term, "Need the keys to use and the term string");
   //yuck full repetition here... fix me...
-  if(!_.isArray(args.keys)){
-    args.keys = [args.keys];
+  if(!_.isArray(args.fields.keys)){
+    args.fields.keys = [args.fields.keys];
   }
   var tsv;
-  if(args.keys.length === 1){
-    tsv = util.format("(body ->> '%s')", args.keys[0]);
+  if(args.fields.keys.length === 1){
+    tsv = util.format("(body ->> '%s')", args.fields.keys[0]);
   }else{
     var formattedKeys = [];
-    _.each(args.keys, function(key){
+    _.each(args.fields.keys, function(key){
       formattedKeys.push(util.format("(body ->> '%s')", key));
     });
     tsv= util.format("concat(%s)", formattedKeys.join(", ' ',"));
   }
+  
   var sql = "select * from " + this.fullname + " where " + util.format("to_tsvector(%s)", tsv);
-  sql+= " @@ to_tsquery($1);";
+  sql+= " @@ to_tsquery($1)";
+  if(args.options && args.options.queryOptions){
```

Comment:
I think this is always going to be true, since searchArgs creates a QueryOptions no matter what and `queryOptions` is a function on that. The default sort will be by the primary key column if the table has one or by `1` (first column in result set) otherwise, so you should be able to just add `args.options.queryOptions()` to the sql no matter what.